### PR TITLE
HAAR-884 FindUserByUsername also calls out to Nomis

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/UserService.kt [hunk 1]
```diff
@@ -13,15 +13,18 @@ class UserService(
...
-      // or nomis user
       ?: run {
-        deliusApiService.findUserByUsername(username)?.toUserDetails()
+        nomisApiService.findUserByUsername(username)?.toUserDetails()
           ?: run {
-            authApiService.findAzureUserByUsername(username)
+            deliusApiService.findUserByUsername(username)?.toUserDetails()
+              ?: run {
+                authApiService.findAzureUserByUsername(username)
+              }
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/UserService.kt:27]: Order is now ext users->nomis->delius->azure

File: src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/UserServiceTest.kt [hunk 2]
```diff
@@ -16,21 +16,24 @@ import uk.gov.justice.digital.hmpps.manageusersapi.adapter.external.UserSearchAp
...
 import uk.gov.justice.digital.hmpps.manageusersapi.model.UserDetailsDto
 import uk.gov.justice.digital.hmpps.manageusersapi.resource.external.ExternalUserDetailsDto
 import java.util.UUID
 
 class UserServiceTest {
-  private val externalUsersApiService: UserSearchApiService = mock()
   private val authApiService: AuthApiService = mock()
   private val deliusUserApiService: UserApiService = mock()
+  private val externalUsersApiService: UserSearchApiService = mock()
+  private val nomisUserApiService: uk.gov.justice.digital.hmpps.manageusersapi.adapter.nomis.UserApiService = mock()
```
Comment (ravmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/UserServiceTest.kt:28]: use import alias :
import com.sorandom.app.model.api.Input as OtherInput

Comment (karenmillermoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/UserServiceTest.kt]: Updated

**Review by ravmoj — APPROVED** [review]
lgtm

**Pull request merged**